packages/eslint-plugin/src/rules/enforce-module-boundaries.ts

```diff
@@ -712,4 +712,4 @@ export default createESLintRule<Options, MessageIds>({
       },
     };
   },
-});
+}) as any;
```

Comment:
How come this needed an `as any`?

Reply:
That was a working version. However, the latest `@typescript-eslint/utils` is no longer inferred so we need to specify it explicitly. Likely, it's something that should be fixed on `@typescript-eslint` side?

Reply:
I've written to you on slack about this